src/Compilers/CSharp/Portable/Binder/Binder_Await.cs

```diff
@@ -185,9 +185,21 @@ private bool ReportBadAwaitWithoutAsync(SyntaxNodeOrToken nodeOrToken, BindingDi
                         }
                         else
                         {
-                            info = method.ReturnsVoid ?
-                                new CSDiagnosticInfo(ErrorCode.ERR_BadAwaitWithoutVoidAsyncMethod) :
-                                new CSDiagnosticInfo(ErrorCode.ERR_BadAwaitWithoutAsyncMethod, method.ReturnType);
+                            if (method.ReturnsVoid)
+                            {
+                                info = new CSDiagnosticInfo(ErrorCode.ERR_BadAwaitWithoutVoidAsyncMethod);
+                            }
+                            else if (method.IsIterator && 
+                                     (method.ReturnType.IsIAsyncEnumerableType(Compilation) || 
+                                      method.ReturnType.IsIAsyncEnumeratorType(Compilation)))
+                            {
+                                // For async iterators, use the generic error that doesn't suggest changing the return type
```

Comment:
```suggestion
                                // For async iterators, use the generic error that doesn't suggest changing the return type
                                // Since the return type is already fine.
```